rust/protocol/src/session_cipher.rs

```diff
@@ -96,6 +104,10 @@ pub async fn message_encrypt(
         )
         .await?
     {
+        log::warn!(
+            "Previously known identity key for while encrypting for recipient: {}",
```

Comment:
I found this phrasing really hard to parse. "for while encrypting"?

How about "Identity key {} is not trusted for remote address {}"?